packages/rxtranscribe/demo/runDemo.js

```diff
@@ -95,29 +105,70 @@ const schema = {
   }
 };
 
+const outputWriter = filePath => message$ => {
```

Comment:
I don't understand the `const variable = a => b => {}` syntax.

Reply:
This is called "currying".  It's when a function returns a function in order to compose a new function that is configured for a specific purpose.  And it's used a lot in functional programming, especially when functions are being combined into a `pipe`. In this example, calling `outputWriter('path/to/file)` returns a function which can be used in an RxJS pipe (and it will write the output to that specific file).

Why do it?  It's often done to create a function that contains some configuration or state that needs to be passed into the internal function. It also tends to simplify unit testing.  And it can be used to manufacture more specific functions. A simple example:
```js
const pipe = input => fns.reduce((acc, fn) => fn(acc), input);
const multiply = num => input => num * input;
const double = multiply(2);
const triple = multiply(3);

const transform = pipe(double, triple);
transform(1); // 1 * 2 * 3 = 6
```
This is the way that RxJS operators and lodash/fp functions work.  So if you want to write your own RxJS operators, they just need to conform to the pattern: `const customOperator = config => source$ => newObservable$;`

It's also useful for unit testing because it simplifies dependency injection:
```js
// I probably don't even need to write a test for this because it's so simple.
// But if you wanted to test it, it would be easy to do so by spying on a fake s3 object:
const saveFile = s3Client => saveParams => s3Client.putObject(s3Params).promise();

// since all of the S3 logic has been pulled out, there's no need for unit or integration testing.  // saveFile could literally be replaced by any function and the only thing to test here 
// is that it's being called.
const performWorkflow({
  s3Params, 
  saveFile = saveFile(new AWS.S3())
}) {
  const promise = saveFile(s3Params);
  return promise;
};

export default performWorkflow;
```